frontend/lib/api/ApiClient.ts

```diff
@@ -12,54 +12,95 @@ export interface ApiResponse<T> {
   data: T;
 }
 
-export type ApiError = {
-  status: ApiResponseStatus.ClientError | ApiResponseStatus.ServerError;
-  code: number;
-  error?: string;
-};
+export class ApiError extends Error {
+  constructor(
+    public status: ApiResponseStatus.ClientError | ApiResponseStatus.ServerError,
+    public code: number,
+    public message: string,
+  ) {
+    super(message || "Unknown error");
```

Comment:
Message is not an optional field, and I do not think you mean to override an empty string? If you do want to make it optional, you can also use default parameter: `public message = "Unknown error"`

Reply:
Check - previously JSON responses from the API were directly cast to `ApiError` that's why there was a error field (and this fallback). But I removed all casting. I'll add your suggestion - thanks!